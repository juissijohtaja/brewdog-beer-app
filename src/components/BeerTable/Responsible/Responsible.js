import React from 'react'
import Avatar from '@material-ui/core/Avatar'
import Box from '@material-ui/core/Box'
import { makeStyles } from '@material-ui/core/styles'
import { grey } from '@material-ui/core/colors'

const useStyles = makeStyles(theme => ({
  avatarResponsible: {
    width: theme.spacing(3.5),
    height: theme.spacing(3.5),
    fontSize: '0.8em',
    fontWeight: 600,
    color: grey[50],
    backgroundColor: theme.palette.secondary.main,
    display: 'inline-flex',
    marginRight: 8,
  },
}))

const Responsible = ({ responsible }) => {
  const classes = useStyles()

  return (
    <Box>
      <Avatar className={classes.avatarResponsible}>
        {responsible.split(' ').map(n => n[0]).join('').substring(0, 2)}
      </Avatar> {responsible}
    </Box>
  )
}

export default Responsible