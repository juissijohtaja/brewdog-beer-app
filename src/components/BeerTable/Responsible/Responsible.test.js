import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import initialStoreState from '../../../setupTests'
import Responsible from './Responsible'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])

describe('Responsible Component', () => {
  let store, component

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()
  })

  test('should render name with one part with one initial', () => {
    component = renderer.create(
      <Provider store={store}>
        <Responsible responsible='Johnny' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render name with two parts with two initial', () => {
    component = renderer.create(
      <Provider store={store}>
        <Responsible responsible='John Doe' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render name with multiple parts with two initials', () => {
    component = renderer.create(
      <Provider store={store}>
        <Responsible responsible='John Doe Old Boy' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})