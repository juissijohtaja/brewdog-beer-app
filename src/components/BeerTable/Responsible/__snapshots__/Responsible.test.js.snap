// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsible Component should render name with multiple parts with two initials 1`] = `
<div
  className="MuiBox-root MuiBox-root-4"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-avatarResponsible-1 MuiAvatar-colorDefault"
  >
    JD
  </div>
   
  John Doe Old Boy
</div>
`;

exports[`Responsible Component should render name with one part with one initial 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-avatarResponsible-1 MuiAvatar-colorDefault"
  >
    J
  </div>
   
  Johnny
</div>
`;

exports[`Responsible Component should render name with two parts with two initial 1`] = `
<div
  className="MuiBox-root MuiBox-root-3"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-avatarResponsible-1 MuiAvatar-colorDefault"
  >
    JD
  </div>
   
  John Doe
</div>
`;
